### PR TITLE
Expand wildcards in snapshot

### LINKED ISSUE #6097: snapshot/restore expand_wildcards option not consistent with other apis
With #4453 we introduced new indices options to be used across all apis that support multiple indices. Those options are listed here in our docs: http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/multi-index.html .

All apis support the `expand_wildcards` option that accepts `open`, `closed` or both as values, while create and restore snapshot apis accept `expand_wildcards_open` and `expand_wildcards_closed` as booleans instead. Looks like we should make them consistent with other apis.

### PR DESCRIPTION
**Pull request by imotov**
This PR makes handling of expand_wildcards parameters consistent. The first commit f5d8ca8 will go to both 1.x and master and the second commit 7abe1ed will go into master only. Closes #6097.

### CODE REVIEW

**Comment by javanna** [comment]
looks good, left a few comments. I think we need to drop the indices options parsing code from multi_search too and reuse this new common method there too.

**Comment by javanna** [comment]
One more thing, I would split this PR in two, the actual change and the breaking part that removes the old parameters, otherwise this whole change ends up in the 1.x release notes as a breaking one, which it isn't.

**Comment by imotov** [comment]
@javanna I push the changes based on your comments. I will open a separate PR for removal of old parameters from 2.0 once this PR is merged.

File: src/main/java/org/elasticsearch/common/xcontent/support/XContentMapValues.java [hunk 1]
```diff
@@ -387,4 +387,23 @@ public static TimeValue nodeTimeValue(Object node) {
...
         }
     }
+
+    /**
+     * Returns an array of string value from a node value.
+     *
+     * If the node represents an array the corresponding array of strings is returned.
+     * Otherwise the node is treated as a comma-separated string.
+     */
+    public static String[] nodeStringArrayValue(Object node) {
```
Comment (javanna) [inline src/main/java/org/elasticsearch/common/xcontent/support/XContentMapValues.java]: I find the `node` wording confusing, it makes me think of elasticsearch nodes. Maybe `xContentNode`?

**Comment by javanna** [comment]
Looks good @imotov , I think we can remove the breaking label? 

It might be good to add some more tests where missing around the apis that this PR touches using the different indices options, just to make sure that we still parse them correctly? I have the feeling that we have no extensive tests around that. What do you think?

File: src/main/java/org/elasticsearch/common/xcontent/support/XContentMapValues.java (hunk 1 above)
Comment (imotov) [inline src/main/java/org/elasticsearch/common/xcontent/support/XContentMapValues.java]: This is a convention of all static methods that are used to parse XContentMapValues.

**Comment by imotov** [comment]
@javanna I pushed additional tests as we discussed. Could you take a look?

File: src/main/java/org/elasticsearch/common/xcontent/support/XContentMapValues.java (hunk 1 above)
Comment (javanna) [inline src/main/java/org/elasticsearch/common/xcontent/support/XContentMapValues.java]: ok sorry I am not too familiar with these methods :)

**Comment by javanna** [comment]
thanks for adding those tests @imotov LGTM

**Pull request merged**